lib/onlyRegex.js

```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+function onlyRegex (pattern) {
+  var match = false
+
+  for (var i = 0; i < Object.keys(pattern).length; i++) {
+    if (pattern[Object.keys(pattern)[i]] instanceof RegExp) {
```

Comment:
`Object.keys()` is really slow, and you are calling it twice in a loop.
You should use https://github.com/mcollina/bloomrun/blob/master/lib/deepMatch.js#L14-L15.